app/services/authentication/authenticator.rb

```diff
@@ -70,7 +76,7 @@ def proper_user(identity)
       end
     end
 
-    def build_user
+    def build_user!
```

Comment:
Changed this method name since we call `save!` not just `save` :)

Reply:
we can also call it `find_existing_user_or_create_it!` or something like that. Not blocking obviously :D